crates/router/tests/connectors/payeezy.rs

```diff
@@ -0,0 +1,761 @@
+            }),
+            None,
+        )
+        .await
+        .unwrap();
+    assert_eq!(
+        *response.response.unwrap_err().message,
+        "Bad Request (25) - Invalid Expiry Date".to_string(),
+    );
+}
+
+// Creates a payment with incorrect expiry year.
+#[actix_web::test]
+async fn should_fail_payment_for_incorrect_expiry_year() {
+    let response = CONNECTOR
+        .make_payment(
+            Some(types::PaymentsAuthorizeData {
+                payment_method_data: types::api::PaymentMethodData::Card(api::Card {
+                    card_exp_year: Secret::new("2000".to_string()),
+                    ..utils::CCardType::default().0
+                }),
+                ..utils::PaymentAuthorizeType::default().0
+            }),
+            None,
+        )
+        .await
+        .unwrap();
+    assert_eq!(
+        response.response.unwrap_err().message,
+        "Expiry Date is invalid".to_string(),
+    );
+}
+
+// Voids a payment using automatic capture flow (Non 3DS).
+#[actix_web::test]
+async fn should_fail_void_payment_for_auto_capture() {
+    let authorize_response = CONNECTOR
+        .authorize_payment(
+            PayeezyTest::get_payment_data(),
+            PayeezyTest::get_payment_info(),
+        )
+        .await
+        .expect("Authorize payment response");
+    let txn_id = utils::get_connector_transaction_id(authorize_response.response.clone()).unwrap();
+    let capture_connector_meta = utils::get_connector_meta(authorize_response.response);
+    let capture_response = CONNECTOR
+        .capture_payment(
+            txn_id,
+            Some(types::PaymentsCaptureData {
+                connector_meta: capture_connector_meta,
+                ..utils::PaymentCaptureType::default().0
+            }),
+            PayeezyTest::get_payment_info(),
+        )
+        .await
+        .expect("Capture payment response");
+    assert_eq!(capture_response.status, enums::AttemptStatus::Charged);
+    let txn_id = utils::get_connector_transaction_id(capture_response.clone().response);
+    let connector_meta = utils::get_connector_meta(capture_response.response);
+    assert_ne!(txn_id, None, "Empty connector transaction id");
+    let void_response = CONNECTOR
+        .void_payment(
+            txn_id.unwrap(),
+            Some(types::PaymentsCancelData {
+                connector_transaction_id: String::from(""),
+                cancellation_reason: Some("requested_by_customer".to_string()),
+                connector_meta,
+                ..Default::default()
+            }),
+            PayeezyTest::get_payment_info(),
+        )
+        .await
+        .expect("Void payment response");
+    assert_eq!(
+        void_response.response.unwrap_err().message,
+        "Bad Request (26) - Invalid Amount"
+    );
+}
+
+// Captures a payment using invalid connector payment id.
+#[actix_web::test]
+async fn should_fail_capture_for_invalid_payment() {
+    let response = PayeezyTest {}
```

Comment:
For this test, you can directly call `capture_payment` with some random `connector_payment_id `